src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java

```diff
@@ -0,0 +1,166 @@
+                BrowseEntity.Builder entity = (BrowseEntity.builder(String.valueOf(object), String.valueOf(object),
+                                ENTITY_TYPE_OBJECTS).
+                        canBrowse(false).canSample(true));
+                browseDetailBuilder.addEntity(entity.build());
+                count++;
+            }
+        } catch (Exception e) {
+            throw new IOException("Unable to create a Connection", e);
+        }
+        return browseDetailBuilder.setTotalCount(count).build();
+    }
+
+    @Override
+    public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+            throws IOException {
+        ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+        Map<String, String> properties = new HashMap<>();
+        properties.put(ConfigUtil.NAME_USE_CONNECTION, "true");
+        properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+        String tableName = connectorSpecRequest.getPath();
+        if (tableName != null) {
+            properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(tableName));
+            properties.put(MarketingCloudConstants.PROPERTY_OBJECT_NAME, SourceObject.valueOf(tableName).getValue());
+            schema = config.getSchema(SourceObject.valueOf(tableName));
+            specBuilder.setSchema(schema);
+        }
+        return specBuilder.addRelatedPlugin(new PluginSpec(MarketingCloudConstants.PLUGIN_NAME, BatchSource.PLUGIN_TYPE,
+                properties)).addRelatedPlugin(new PluginSpec(MarketingCloudDataExtensionSink.PLUGIN_TYPE,
+                BatchSink.PLUGIN_TYPE, properties)).build();
+    }
+
+    @Override
+    public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+            throws IOException {
+        String object = sampleRequest.getPath();
+        if (object == null) {
+            throw new IllegalArgumentException("Path should contain object.");
+        }
+        try {
+            return listObjectDetails(SourceObject.valueOf(object));
+        } catch (Exception e) {
+            throw new IOException("unable to fetch records", e);
+        }
+    }
+
+    private List<StructuredRecord> listObjectDetails(SourceObject sourceObject) throws ETSdkException
+            , IOException {
+        List<StructuredRecord> sample = new ArrayList<>();
+        MarketingCloudClient marketingCloudClient = MarketingCloudClient.create
+                (config.getClientId(), config.getClientSecret(),
+                        config.getAuthEndpoint(), config.getSoapEndpoint());
+        if (sourceObject == SourceObject.DATA_EXTENSION) {
+            return Collections.emptyList();
+        } else {
+            response = marketingCloudClient.fetchObjectRecords(sourceObject, null);
+            results = response.getObjects();
+            iterator = results.iterator();
```

Comment:
iterator = response.getObjects().iterator(); use this  and remove results variable from top

Reply:
Changed it and remove the above result line